hsc/crawler.py

```diff
@@ -113,26 +148,28 @@ def get_submissions(self, submissions):
 					track_folder_name = track['name'].strip().replace(' ', '')
 					track_url = self.domain_url.format(track['track_slug'], track['slug'])
 					parent_folder_name = track['track_name'].strip().replace(' ', '')
-					folder_name = os.path.join(parent_folder_name ,track_folder_name)
-				
-				if language in self.special_extensions:
-					file_extension = '.' + self.special_extensions[language]
+					folder_name = os.path.join(parent_folder_name, track_folder_name)
+
+				if self.make_language_folder:
+					folder_name = os.path.join(folder_name, language)
+
+				if language in self.file_extensions:
```

Comment:
What if a language is not listed in our file_extensions map? Will there be no extension to that file name? I think we should at least have `language` as the file_extension in that case. Otherwise, there will be no way to judge the language of that file.

Reply:
The default file extension is '.'+language

Reply:
https://github.com/Nullifiers/Hackerrank-Solution-Crawler/blob/add_file_extensions/hsc/crawler.py#L142

Reply:
Ohh, totally missed that. Thanks :+1: